proxy/src/lib.rs

```diff
@@ -410,10 +414,15 @@ where
     let accept = bound_port.listen_and_fold(
         (),
         move |(), (connection, remote_addr)| {
-            server.serve(connection, remote_addr);
-            Ok(())
+            let s = server.serve(connection, remote_addr);
+            let s = ::logging::context_future((name, remote_addr), s);
+            let r = DefaultExecutor::current()
+                .spawn(Box::new(s))
```

Comment:
Would this be better done by using `tokio_executor::with_default` to replace the default executor with a custom `LoggingExecutor` that does, or equivalent?

Also, shouldn't this be done at a much higher layer? What if `listen_and_fold` itself executes futures that need logging, before *this* future runs, which indeed will probably be the case soon with TLS.

Reply:
@briansmith I agree with your comment about using `context_executor` rather than `context_future`, and will change that. However, I'm not sure if I understand why `listen_and_fold` would ever need to execute any futures rather than returning one chained future?

Reply:
Ah, I misunderstood what you were saying, I thought  you meant that `listen_and_fold` was going to _spawn_ futures prior to returning one. In that case, you're correct that this `context_future` won't include anything that happens before the `fold`, however, it's constructed here because  in this context, we have the remote address. The whole future returned by `listen_and_fold` is wrapped in an outer  context which (currently) has only the name of which side of the proxy it corresponds to: https://github.com/runconduit/conduit/pull/1025/commits/11e561553b1b30416edb03f93044e167c4f2849e#diff-2337e2772bb9d18416c468ce781c8d56R425

Reply:
OK, gotcha. Thanks for the explanation. Then in the listen_and_fold case, as part of my TLS work series of PRs, I'll move this logic into listen_and_fold itself, right after we acquire the remote address, and before we do the TLS handshake.

Reply:
Great, that sounds right to me. Thanks!